### PR TITLE
DCM-24: Remove unsupported mappings from automation view

### PR DESCRIPTION
**Pull request by anjula-sack**
### The issue I worked on
See https://issues.openmrs.org/browse/DCM-24

### Description of what I changed:
Add an if condition to check the period types

I have used the period type as the name of the mapping when testing :

image

### CODE REVIEW

File: omod/src/main/webapp/automation.jsp [hunk 1]
```diff
@@ -33,7 +36,15 @@
                     <select name="mapping">
                         <option></option>
                         <c:forEach items="${mappings}" var="mapping">
-                            <option value="${mapping.name}.${mapping.created}">${mapping.name}</option>
+                            <c:choose>
+                                <c:when test="${mapping.periodType == 'Daily' ||
+                            mapping.periodType == 'Weekly' || mapping.periodType == 'Monthly' || mapping.periodType == 'Yearly'}">
```
Comment (jayasanka-sack) [inline omod/src/main/webapp/automation.jsp:41]: Shall we create an array instead?

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Thank you very much Anjual! This is awesome!

I've added a comment.

Also, change the PR title and the commit message with what you did.

File: omod/src/main/webapp/automation.jsp (hunk 1 above)
Comment (anjula-sack) [reply omod/src/main/webapp/automation.jsp]: I filtered the array in java

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java [hunk 2]
```diff
@@ -353,13 +354,15 @@ public void renderAutomationPage(ModelMap model) {
 	}
 
 	private void initialiseAutomation(ModelMap model, boolean automationEnabled, List<String> postResponse) {
+		List<String> supportedPeriodTypes = Arrays.asList("Daily", "Weekly", "Monthly", "yearly", "Financial April",
+				"Financial July", "Financial Oct");
 		List<DHISOrganisationUnit> orgUnits = Context.getService(DHISConnectorService.class).getDHISOrgUnits();
 
-		model.addAttribute("mappings", Context.getService(DHISConnectorService.class).getMappings());
+		model.addAttribute("mappings", Context.getService(DHISConnectorService.class).getMappings()
+				.stream().filter(mapping -> supportedPeriodTypes.contains(mapping.getPeriodType())).collect(Collectors.toList()));
```
Comment (jayasanka-sack) [inline omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java:362]:
```suggestion
		// add a comment
		List<type> supportedMappings = Context.getService(DHISConnectorService.class).getMappings()
						.stream()
						.filter(mapping -> supportedPeriodTypes.contains(mapping.getPeriodType()))
						.collect(Collectors.toList()));	
		model.addAttribute("mappings", supportedMappings); 
```

Note the formatting.

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java [hunk 3]
```diff
@@ -353,13 +354,15 @@ public void renderAutomationPage(ModelMap model) {
 	}
 
 	private void initialiseAutomation(ModelMap model, boolean automationEnabled, List<String> postResponse) {
+		List<String> supportedPeriodTypes = Arrays.asList("Daily", "Weekly", "Monthly", "yearly", "Financial April",
+				"Financial July", "Financial Oct");
```
Comment (jayasanka-sack) [inline omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java:358]:
```suggestion
		List<String> supportedPeriodTypes = Arrays.asList(
				"Daily", 
				"Weekly", 
				"Monthly", 
				"yearly", 
				"Financial April",
				"Financial July", 
				"Financial Oct");
```

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Well done @anjula-sack!

Minor request, Just fix the formattings as suggested. 
Update the commit message too

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java (hunk 3 above)
Comment (Piumal1999) [inline omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java:358]: `"yearly"` - is this case sensitive?

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java [hunk 4]
```diff
@@ -353,13 +355,24 @@ public void renderAutomationPage(ModelMap model) {
 	}
 
 	private void initialiseAutomation(ModelMap model, boolean automationEnabled, List<String> postResponse) {
+		List<String> supportedPeriodTypes = Arrays.asList(
```
Comment (jayasanka-sack) [inline omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java:358]: Let's make this a constant vatiable

**Review by jayasanka-sack — APPROVED** [review]
Thanks @anjula-sack !

**Pull request merged**